### ISSUE TITLE
ZeroNet is going to be licensed as GPLv3+? Great news!

### ISSUE THREAD

**Issue by AnonymusChronicler** [AnonymusChronicler/Anonymus#1]
Interesting: https://github.com/HelloZeroNet/ZeroNet/issues/2273 :tada:

**Comment by 0x6a73** [comment]
Creating more accounts won't change anything. I've reported you again, hopefully the Github Support team will be faster this time.

Also, I'm still waiting for you on the #zeronet-drama channel, as stated previously.

**Comment by 0x6a73** [comment]
Also, nope. It's going to be split into two different repos, one licensed with GPLv3 and the other licensed with some lax license. Just to clarify.

**Comment by 0x6a73** [comment]
Creating more accounts won't change anything. I've reported you again, hopefully the Github Support team will be faster this time.

Also, I'm still waiting for you on the #zeronet-drama channel, as stated previously.

**Comment by AnonymusChronicler** [comment]
> 
>     I don't care if you accept switching to whatever issue those who run ZeroNet project want. This is the same as the GPLv3 and Lax option below but it might include more licenses if it's found appropriate
>     GPLv3 and Lax if you accept switching to either GPLv3 or later or to GPLv3-only or a Lax/Permissive license
>     GPLv3+ and Lax if you accept switching to either GPLv3 or later or a Lax/Permissive license
>     GPLv3-only and Lax if you accept switching to either GPLv3-only or a Lax/Permissive license
>     GPLv3+ if you accept switching to GPLv3 or later
>     GPLv3-only if you accept switching to GPLv3-only
>     GPLv3 if you accept switching to either GPLv3 or later or to GPLv3-only
>     AGPLv3 if you accept switching to AGPLv3 (this would require modified ZeorNet proxies to release their modified source code)
>     MIT/BSD2 if you accept switching to MIT or BSD-2-Clause
>     BSD3 if you accept switching to BSD-3-Clause
>     Apache2 if you accept switching to an Apache-2.0
>     Lax if you accept switching to any Lax/Permissive license (except Public Domain)
>     None if you don't accept changing license, or if you want a different license not listed
> 

:thinking: What is this? 
2 times mentioned GPLv3+ 
5 times mantioned GPLv3-only
How can be this called a fair `voting`? :thinking:

**Comment by 0x6a73** [comment]
Creating more accounts won't change anything. I've reported you again, hopefully the Github Support team will be faster this time.

Also, I'm still waiting for you on the #zeronet-drama channel, as stated previously.

**Comment by 0x6a73** [comment]
Creating more accounts won't change anything. I've reported you again, hopefully the Github Support team will be faster this time.

Also, I'm still waiting for you on the #zeronet-drama channel, as stated previously.

**Comment by 0x6a73** [comment]
Creating more accounts won't change anything. I've reported you again, hopefully the Github Support team will be faster this time.

Also, I'm still waiting for you on the #zeronet-drama channel, as stated previously.

**Comment by AnonymusChronicler** [comment]
@0x6a73 Can you answer my question about the fair voting?

**Comment by 0x6a73** [comment]
Of course. We gave an option to vote for GPLv3+ and other licenses, so I don't see a single problem. It's clearly listed out there.

**Comment by 0x6a73** [comment]
Creating more accounts won't change anything. I've reported you again, hopefully the Github Support team will be faster this time.

Also, I'm still waiting for you on the #zeronet-drama channel, as stated previously.

**Comment by 0x6a73** [comment]
Creating more accounts won't change anything. I've reported you again, hopefully the Github Support team will be faster this time.

Also, I'm still waiting for you on the #zeronet-drama channel, as stated previously.

**Comment by AnonymusChronicler** [comment]
> Of course. We gave an option to vote for GPLv3+ and other licenses, so I don't see a single problem. It's clearly listed out there.

I just [see a new post from @filips123](https://github.com/HelloZeroNet/ZeroNet/issues/2273#issuecomment-549986848) and he is already decided that GPLv3-only and "lax" will be the license while the voting results clearly indicates the winner is GPLv3+! 

Are [this post](https://github.com/HelloZeroNet/ZeroNet/issues/2273#issuecomment-549986848) by @filips123 indicates that ZeroNet doesn't respect what the contributors want?

**Comment by 0x6a73** [comment]
Creating more accounts won't change anything. I've reported  you again, hopefully the Github Support team will be faster this time. Also, I'm still waiting for you on the #zeronet-drama channel on Rizon, as stated previously.

**Comment by 0x6a73** [comment]
@AnonymusChronicler This is a multiple choice voting. Most people voted for "GPLv3 and Lax", yet the statistics show these separately.

**Comment by 0x6a73** [comment]
Creating more accounts won't change anything. I've reported  you again, hopefully the Github Support team will be faster this time. Also, I'm still waiting for you on the #zeronet-drama channel on Rizon, as stated previously.

**Comment by imachug** [comment]
> I just see a new post from @filips123 and he is already decided that GPLv3-only and "lax" will be the license while the voting results clearly indicates the winner is GPLv3+!

Ahem... You do understand GPLv3 is different from GPLv3+ and GPLv3-only yeah?

**Comment by AnonymusChronicler** [comment]
@0x6a73 @filips123 this is your professional answer to a legitimate question about the voting? Not like I will downgrade my license! I have in my repository GPLv3+ on ZeroNet! 

It would be stupid to license it as GPL.3.0-only! What if in a few year there is GPLv4? Than voting again? So stupid. I'm against it! 

Why? :thinking:
### **ZeroNet should be licensed under the most permissive license!**

**Comment by 0x6a73** [comment]
Great. That's why we're licensing most of it under the MIT license.

**Comment by 0x6a73** [comment]
Creating more accounts won't change anything. I've reported you again, hopefully the Github Support team will be faster this time. Also, I'm still waiting for you on the #zeronet-drama channel on Rizon, as stated previously.

**Comment by filips123** [comment]
> I have in my repository GPLv3+ on ZeroNet!

Yes, sure. If you read my post, you will see that there will be some period when ZeroNet will be licensed under GPLv3 (-only or -or-later). After modularization is done, part of it (core libraries) will be re-licensed to MIT but the whole program stays GPLv3 (-only or -or-later).

If you will want to keep only GPLv3 (-only or -or-later), you will still be able to host and distribute it with code from that period. If you would want to use those new versions, you will have to accept to MIT + GPLv3 (-only or -or-later) licenses.

> ZeroNet should be licensed under the most permissive license!

Do you know that MIT is one of the most permissive licenses?

**Comment by AnonymusChronicler** [comment]
> > I just see a new post from @filips123 and he is already decided that GPLv3-only and "lax" will be the license while the voting results clearly indicates the winner is GPLv3+!
> 
> Ahem... You do understand GPLv3 is different from GPLv3+ and GPLv3-only yeah?

You are wrong @imachug please look at this and tell to @filips123 also:

When you say `GPLv3` you mean: **gpl-3.0**, https://enterprise.dejacode.com/licenses/public/gpl-3.0/ 
When you say `GPLv3+`you mean: **gpl-3.0-plus**, https://enterprise.dejacode.com/licenses/public/gpl-3.0-plus/
![difference cleaned](https://user-images.githubusercontent.com/57413853/68241745-27292380-0007-11ea-86e0-48ac34900dd5.png)

**There is only two options!** 

In **both** cases you have automatically Lax! There is no need to vote about that!

**Comment by 0x6a73** [comment]
Nope, when you select the GPLv3 license, it's not easily relicense-able to a lax license. Also, learn what "lax" means.

**Comment by filips123** [comment]
@AnonymusChronicler Read what main pool post says:

> GPLv3+ if you accept switching to GPLv3 or later
> GPLv3-only if you accept switching to GPLv3-only
> GPLv3 if you accept switching to either GPLv3 or later or to GPLv3-only

That's what I was reffering for.

**Comment by AnonymusChronicler** [comment]
I know exactly what that means but you saying; 
> it's not easily relicense-able to a lax license. 

Do you understand that there is no way to revoke your rights which once granted under GPLv3 and GPLv3+? If you anyway try to restrict the use of the software ZeroNet repository will loose the following: 

**This license may end automatically if you bring a patent claim against any contributor over patents that you claim are infringed by the licensed software; your patent license from such contributor to the licensed software may also end automatically.**

**Comment by 0x6a73** [comment]
We won't repeat ourselves again. Contributors can relicense their own code as they see fit, according to the copyright law of EU and USA.

**Comment by filips123** [comment]
> Do you understand that there is no way to revoke your rights which once granted under GPLv3 and GPLv3+?

Nobody will revoke your rights. Old versions will remain licensed completely as GPl. New versions will be licensed one part as GPl and one as MIT.

> If you anyway try to restrict the use of the software ZeroNet repository will loose the following

Again. Relicensing to MIT or similar license would make ZeroNet less restricted

**Comment by AnonymusChronicler** [comment]
@filips123 That is not going to happen! The license either GPLv3 or nothing! You have no choice at all! The @HelloZeroNet repository in this very moment still infringes other people rights both in licenses and copyright. You must upgrade the license to GPLv3+ preferable over GPLv3 and all licensing issues solved! 

You can always use `LINKING EXCEPTION` in the `LICENSE` file but **if I find that the linking exception is wrong**, I as a downstream **can throw that exception away** this applies to both GPLv3 and GPLv3+!

**Comment by filips123** [comment]
> That is not going to happen!

Why?

> You have no choice at all! The @HelloZeroNet repository in this very moment still infringes other people rights both in licenses and copyright. 

Read again: https://github.com/AnonymusChronicler/Anonymus/issues/1#issuecomment-549994996

**Comment by 0x6a73** [comment]
@AnonymusChronicler Learn to read.

**Comment by AnonymusChronicler** [comment]
> We won't repeat ourselves again. Contributors can relicense their own code as they see fit, according to the copyright law of EU and USA.

WHAT? :rofl: 

### "Each contributor grants you a non-exclusive, worldwide, royalty-free patent license under the contributor's essential patent claims, to make, use, sell, offer for sale, import and otherwise run, modify and propagate the contents of its contributor version."

**Comment by 0x6a73** [comment]
Also, at this point I'll stop posting in these forks, limiting myself to just reporting them. I hope that someone else will be able to  eep mertitorical discussion up.

**Comment by 0x6a73** [comment]
@AnonymusChronicler Oh, and my last comment: USA/EU law is above licenses.

**Comment by AnonymusChronicler** [comment]
@filips123 @imachug If this @HelloZeroNet repository on GitHub is not open for pull requests than those who distribute they own version will have they own copyright on the software! 

Each change must be recorded in a `ChangeLog` so you will not own any rights over those changes and can't incorporate it into your version without giving credit!

@shortcutme has a mania to steal from others and merge pull request in his name. THis can't be happen from now on. I own also right in ZeroNet and worth noting that copyright is not measured in commits! 

Right now in **each of you have my code** in ZeroNet and can see the result when visiting Ui or any other pages ZeroNet!

**Comment by xTibor** [comment]
Meow meow meow, meow meow. Meow meow meow. Purr, purr. Meow meow meow, meow meow. Meow meow meow. Purr, purr. Purr purr purr. Meow. Meow meow. Purr purr. Meow. Meow meow.

![Cat](http://i.imgur.com/q6HH0.jpg)

Purr purr purr. Meow purr. Purr, meow.

![Cat](http://1.bp.blogspot.com/-b4IV6-4UKpE/UK0DtDY7S9I/AAAAAAAAbX0/-tF21BSsTXY/s1600/funny-cat-pictures-025-010.jpg)

Meow meow meow meow, meow meow. Purr. Purr purr. Meow meow meow, meow meow. Meow meow meow. Purr, purr. Purr purr purr. Meow. Meow meow. Meow meow meow.

**Comment by AnonymusChronicler** [comment]
@xTibor @HelloZeroNet (@shortcutme)
![PICSA](https://user-images.githubusercontent.com/57413853/68251272-a8d67c80-001a-11ea-8966-de481a22f186.png)
# Magyar Népköztársaság
<a href="http://www.youtube.com/watch?feature=player_embedded&v=u1x6nuEfyCM" target="_blank"><img src="http://img.youtube.com/vi/u1x6nuEfyCM/0.jpg" alt="HétköznaPI CSAlódások - Magyar Népköztársaság" width="max" height="420" border="10" align="center"/></a>

**Comment by AnonymusChronicler** [comment]
https://github.com/HelloZeroNet/ZeroNet/issues/2273

**Comment by bela-miklos** [comment]
![https://i.ytimg.com/vi/zywYHyQA0rg/maxresdefault.jpg](https://i.ytimg.com/vi/zywYHyQA0rg/maxresdefault.jpg)

**Comment by 0x6a73** [comment]
Yes

**Comment by 0x6a73** [comment]
Remember sir to do the needful sir
sir
![dotheneedful](https://user-images.githubusercontent.com/43107265/68307127-a91a5a80-00a2-11ea-9cea-fea588089398.png)
